models.py

```diff
@@ -99,219 +83,131 @@ def conv_block(input_tensor, kernel_size, filters, stage, block, strides=(2, 2))
+    x = MaxPooling2D((3, 3) , data_format=data_format , strides=(2, 2))(x)
     
 
-    x = conv_block(x, 3, [64, 64, 256], stage=2, block='a', strides=(1, 1))
-    x = identity_block(x, 3, [64, 64, 256], stage=2, block='b')
-    x = identity_block(x, 3, [64, 64, 256], stage=2, block='c')
-    f2 = one_side_pad(x )
+    x = conv_block(x, 3, [64, 64, 256], stage=2, block='a', strides=(1, 1), data_format=data_format)
+    x = identity_block(x, 3, [64, 64, 256], stage=2, block='b', data_format=data_format)
+    x = identity_block(x, 3, [64, 64, 256], stage=2, block='c', data_format=data_format)
+    f2 = ZeroPadding2D(((1,0), (1,0)), data_format=data_format)(x)
 
 
-    x = conv_block(x, 3, [128, 128, 512], stage=3, block='a')
-    x = identity_block(x, 3, [128, 128, 512], stage=3, block='b')
-    x = identity_block(x, 3, [128, 128, 512], stage=3, block='c')
-    x = identity_block(x, 3, [128, 128, 512], stage=3, block='d')
+    x = conv_block(x, 3, [128, 128, 512], stage=3, block='a', data_format=data_format)
+    x = identity_block(x, 3, [128, 128, 512], stage=3, block='b', data_format=data_format)
+    x = identity_block(x, 3, [128, 128, 512], stage=3, block='c', data_format=data_format)
+    x = identity_block(x, 3, [128, 128, 512], stage=3, block='d', data_format=data_format)
     f3 = x 
 
-    x = conv_block(x, 3, [256, 256, 1024], stage=4, block='a')
-    x = identity_block(x, 3, [256, 256, 1024], stage=4, block='b')
-    x = identity_block(x, 3, [256, 256, 1024], stage=4, block='c')
-    x = identity_block(x, 3, [256, 256, 1024], stage=4, block='d')
-    x = identity_block(x, 3, [256, 256, 1024], stage=4, block='e')
-    x = identity_block(x, 3, [256, 256, 1024], stage=4, block='f')
+    x = conv_block(x, 3, [256, 256, 1024], stage=4, block='a', data_format=data_format)
+    x = identity_block(x, 3, [256, 256, 1024], stage=4, block='b', data_format=data_format)
+    x = identity_block(x, 3, [256, 256, 1024], stage=4, block='c', data_format=data_format)
+    x = identity_block(x, 3, [256, 256, 1024], stage=4, block='d', data_format=data_format)
+    x = identity_block(x, 3, [256, 256, 1024], stage=4, block='e', data_format=data_format)
+    x = identity_block(x, 3, [256, 256, 1024], stage=4, block='f', data_format=data_format)
     f4 = x 
 
-    x = conv_block(x, 3, [512, 512, 2048], stage=5, block='a')
-    x = identity_block(x, 3, [512, 512, 2048], stage=5, block='b')
-    x = identity_block(x, 3, [512, 512, 2048], stage=5, block='c')
+    x = conv_block(x, 3, [512, 512, 2048], stage=5, block='a', data_format=data_format)
+    x = identity_block(x, 3, [512, 512, 2048], stage=5, block='b', data_format=data_format)
+    x = identity_block(x, 3, [512, 512, 2048], stage=5, block='c', data_format=data_format)
     f5 = x 
 
-
     if pretraining:
-        model=Model( img_input , x ).load_weights(resnet50_Weights_path)
+        Model(img_input, x).load_weights(resnet50_Weights_path)
 
+    if light_version:
```

Comment:
Duplicate code complicates code maintenance, so I merged the light version in here.